src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/9.0-1.sql

```diff
@@ -73,22 +73,16 @@ BEGIN
     FROM pg_catalog.pg_dist_partition p;
 
     -- restore pg_dist_object from the stable identifiers
-    WITH old_records AS (
-        DELETE FROM
-            citus.pg_dist_object
-        RETURNING
-            type,
-            object_names,
-            object_args
-    )
-    INSERT INTO citus.pg_dist_object (classid, objid, objsubid)
-    SELECT
-        address.classid,
-        address.objid,
-        address.objsubid
-    FROM
-        old_records naming,
-        pg_get_object_address(naming.type, naming.object_names, naming.object_args) address;
+    UPDATE citus.pg_dist_object
+    SET classid = address.classid, objid = address.objid, objsubid = address.objsubid
```

Comment:
We didn't do an update here because it can cause a primary key violation when a new OID matches an old OID. We also couldn't use deferred constraints in combination with systable functions.